pyccel/ast/numpyext.py

```diff
@@ -2310,6 +2313,116 @@ def __new__(cls, a, axis = None):
 
         return PyccelArrayShapeElement(a, axis)
 
+class NumpyIsNan(NumpyUfuncBase):
```

Comment:
Why did you not subclass `NumpyUfuncUnary` instead?

Reply:
No obvious reason. I also suggested subclassing `NumpyUfuncUnary` originally. I have fixed this [679451a](https://github.com/pyccel/pyccel/pull/1552/commits/679451a9f835a7471e59e2b6307bd2bd80f89962)